library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -0,0 +1,652 @@
+      // If it is a replay state, don't start any threads here
+      return;
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      if (dbPoller != null && dbPoller.isAlive()) {
+        dbPoller.interrupt();
+        dbPoller.join();
+      }
+    } catch (InterruptedException ex) {
+      // log and ignore, ending execution anyway
+      LOG.error("exception in poller thread: ", ex);
+    }
+  }
+
+  @Override
+  public void handleIdleTime()
+  {
+    if (execute) {
+      try {
+        Thread.sleep(spinMillis);
+      } catch (InterruptedException ie) {
+        throw new RuntimeException(ie);
+      }
+    } else {
+      LOG.error("Exception: ", cause);
+      DTThrowable.rethrow(cause.get());
+    }
+  }
+
+  protected void replay(long windowId) throws SQLException
+  {
+    isReplayed = true;
+
+    MutablePair<String, String> recoveredData = new MutablePair<String, String>();
+    try {
+      recoveredData = (MutablePair<String, String>)windowManager.load(operatorId, windowId);
+
+      if (recoveredData != null) {
+        //skip the window and return if there was no incoming data in the window
+        if (recoveredData.left == null || recoveredData.right == null) {
+          return;
+        }
+
+        if (recoveredData.right.equals(rangeQueryPair.getValue()) || recoveredData.right.equals(previousUpperBound)) {
+          LOG.info("Matched so returning");
+          return;
+        }
+
+        JdbcPollInputOperator jdbcPoller = new JdbcPollInputOperator();
+        jdbcPoller.setStore(store);
+        jdbcPoller.setKey(getKey());
+        jdbcPoller.setPartitionCount(getPartitionCount());
+        jdbcPoller.setPollInterval(50);
+        jdbcPoller.setTableName(getTableName());
+        jdbcPoller.setBatchSize(getBatchSize());
+        isPollable = false;
+
+        LOG.debug("[Window ID -" + windowId + "," + recoveredData.left + "," + recoveredData.right + "]");
+
+        jdbcPoller.setRangeQueryPair(new KeyValPair<String, String>(recoveredData.left, recoveredData.right));
+
+        jdbcPoller.ps = jdbcPoller.store.getConnection().prepareStatement(
+            JdbcMetaDataUtility.buildRangeQuery(jdbcPoller.getTableName(), jdbcPoller.getKey(),
+                jdbcPoller.getRangeQueryPair().getKey(), jdbcPoller.getRangeQueryPair().getValue()),
+            java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+        LOG.info("Query formed for recovered data -" + jdbcPoller.ps.toString());
+
+        emitReplayedTuples(jdbcPoller.ps);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("replay", e);
+    }
+
+  }
+
+  /**
+   * Replays the tuples in sync mode for replayed windows
+   */
+  public void emitReplayedTuples(PreparedStatement ps)
+  {
+    LOG.debug("Emitting replayed statement is -" + ps.toString());
+    ResultSet rs = null;
+    try (PreparedStatement pStat = ps;) {
+      pStat.setFetchSize(10000);
```

Comment:
Can this be picked up from some property. Else declare it as a constant at the top.